modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/redshift/Main.scala

```diff
@@ -0,0 +1,142 @@
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.redshift
+
+import cats.Applicative
+import cats.data.Validated._
+import cats.effect.{Concurrent, ExitCode, IO, IOApp, Timer}
+import cats.implicits._
+
+import com.snowplowanalytics.snowplow.rdbloader.common.Common
+import com.snowplowanalytics.snowplow.rdbloader.common.TableDefinitions.AtomicDefaultColumns
+import com.snowplowanalytics.snowplow.rdbloader.core.config.CliConfig
+import com.snowplowanalytics.snowplow.rdbloader.core.LoaderError
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Migration
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Manifest
+import com.snowplowanalytics.snowplow.rdbloader.core.discovery.DataDiscovery
+import com.snowplowanalytics.snowplow.rdbloader.core.algebras._
+import com.snowplowanalytics.snowplow.rdbloader.redshift.loading.Load
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Statement.{
+  CreateAlertingTempTable,
+  DropAlertingTempTable,
+  FoldersCopy,
+  FoldersMinusManifest
+}
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+import fs2.Stream
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import com.snowplowanalytics.snowplow.rdbloader.core.config.StorageTarget.Redshift
+import com.snowplowanalytics.snowplow.rdbloader.core.algebras.FolderMonitoring.TempTableOps
+
+object Main extends IOApp {
+
+  implicit private val LoggerName =
+    Logging.LoggerName(getClass.getSimpleName.stripSuffix("$"))
+
+  def run(argv: List[String]): IO[ExitCode] =
+    CliConfig.parse(argv) match {
+      case Valid(cli) =>
+        Environment.initialize[IO](cli).use { env =>
+          import env._
+
+          Logging[IO].info(s"RDB Loader ${BuildInfo.version} has started. Listening to ${cli.config.messageQueue}") *>
+            process[IO](cli, control).compile.drain.as(ExitCode.Success).handleErrorWith(handleFailure[IO])
+        }
+      case Invalid(errors) =>
+        val logger = Slf4jLogger.getLogger[IO]
+        logger.error("Configuration error") *>
+          errors.traverse_(message => logger.error(message)).as(ExitCode(2))
+    }
+
+  /**
+    * Main application workflow, responsible for discovering new data via message queue
+    * and processing this data with loaders
+    *
+    * @param cli whole app configuration
+    * @param control various stateful controllers
+    * @return endless stream waiting for messages
+    */
+  def process[F[_]: Concurrent: AWS: Iglu: Cache: Logging: Timer: Monitoring: JDBC](
```

Comment:
TableOps algebra would remove most of code here:
1) It will be always redshift, so need to match
2) def process[F[_]: ...: TableOps] (..) ( val folderMonitoring: Stream[F, Unit] = FolderMonitoring.run[F](cli.config.monitoring.folders ... )
3) run in FolderMonitoring also gets [F[_]:TableOps] in type